crates/biome_js_analyze/src/lint/nursery/no_react_specific_props.rs

```diff
@@ -0,0 +1,101 @@
+use biome_analyze::context::RuleContext;
+use biome_analyze::{declare_rule, ActionCategory, Ast, Rule, RuleDiagnostic, RuleSource};
+use biome_console::markup;
+use biome_diagnostics::Applicability;
+use biome_js_factory::make::{jsx_ident, jsx_name};
+use biome_js_syntax::{AnyJsxAttributeName, JsxAttribute};
+use biome_rowan::{AstNode, BatchMutationExt, TextRange};
+
+use crate::JsRuleAction;
+
+declare_rule! {
+    /// Prevents React-specific JSX properties from being used.
+    ///
+    /// This rule is intended for use in JSX-based frameworks (mainly Solid.js)
+    /// that do not use React-style prop names.
+    ///
+    /// ## Examples
+    ///
+    /// ### Invalid
+    ///
+    /// ```js,expect_diagnostic
+    /// <Hello className="John" />
+    /// ```
+    ///
+    /// ### Valid
+    ///
+    /// ```js
+    /// <Hello class="Doe" />
+    /// ```
+ pub NoReactSpecificProps {
+        version: "1.0.0",
+        name: "noReactSpecificProps",
+        sources: &[RuleSource::EslintSolid("no-react-specific-props")],
+        recommended: false,
+    }
+}
+
+const REACT_SPECIFIC_JSX_PROPS: &[&str] = &["className", "htmlFor"];
+
+fn get_replacement_for_react_prop(str: &str) -> &'static str {
+    match str {
+        "className" => "class",
+        "htmlFor" => "for",
+        _ => panic!("should be a React prop: {str:?}"),
```

Comment:
`panic!` like this isn't a good practice, and it's better to use a safer approach. I suggest: to return a `Option<&'static str>`, and instead of `panic` to return a `None`. And return `Some("class")`

Then, at line 61, you call the function like this: `get_replacement_for_react_prop(&name)?`